### ISSUE TITLE
Render resource geojson data

### ISSUE THREAD

**Issue by rufuspollock** [datopian/frontend-showcase-js#122]
When we get a resource with geojson data we should render it as a simple map rather than a table.

Some examples:

http://data.okfn.org/data/core/geo-admin1-us
http://data.okfn.org/data/core/geo-nuts-administrative-boundaries

## Tasks

Research:

- [x] Clarify what type of data we want to render
  - [x] only point and Multi Polygon?
- [x] Explore which library we have to use
  - [x] Is plotly good enough ?
  - [x] What are the other options?
- [x] Come up with recommendation on what we use.
- [x] Explore How the old code in data.okfn.org has been implemented
  - [ ] Answer: backend checks if geojson uses reclinejs

Implementation:

- [x] Write a test using sample data packages (locally)
  - [x] Find sample data
    - [x] https://github.com/frictionlessdata/example-geojson
    - [x] https://github.com/frictionlessdata/example-topojson
  - [x] Write render test for a LeafletMap component that is a thin wrapper around Leaflet (this will not exist yet => failing test at this point)
- [x] Start developing the component to stop that test failing
  - [x] Is there a react + leaflet component out there - we can use or use it as a guide
    - [x] https://github.com/PaulLeCam/react-leaflet
    - [x] Also see old recline code https://github.com/okfn/recline/blob/master/src/view.map.js
- [x] resource -> [map spec] -> leaflet config
  - [x] Generate specs for leafletjs from datapackage.json resource item
  - [x] Hook in to code that generates resource previews to handle geojson or topojson
- [x] Integrate with dpr-api (if any integration needed)

## Acceptance Criteria

- [x] A resource with geojson data renders properly in data package details page.
- [ ] A resource with topojson data renders properly (optional - only do if easy)

### Requirements

We want to render geojson data in the data package view page.

Reference: http://frictionlessdata.io/guides/publish/geo/

What I got from above reference is that we mainly support 

- Multi Polygon
- Point

It can be mix and match of both.

#### Is plotly good enough?

We have plotly in our dependency list already. So it would be great if we can use it for geo json render also.

In `plotly.js` we have some support for Multi Polygon but not points and polygon etc. Here are two issues in plotly:

- https://github.com/plotly/plotly.js/issues/249 
  - Plotting in a city map is not very easy (and limited functionality)
- https://github.com/plotly/plotly.js/issues/326
  - plotly.js maps don't support GeoJSON data nor city-level basemaps at the moment. They recommend leaflet

In summary:

- Plotly does not support `GeoJson` directly
- mix and match of Multi Polygon and point is not possible
- City level plotting is not possible

#### What are the other options?

[leafletjs](http://leafletjs.com/examples/geojson/) is really good choice as it directly support geojson. It has support for:

- Point
- Line String
- Polygon
- Multi Point
- Multi Line String
- Multi Polygon
- Geometry Collection. 

And we can mix and match all types of data.

#### Come up with recommendation on what we use

Clearly leafletjs is winner between the two

**Comment by subhankarb** [comment]
Updated the summary with the results of research: we should use leaflet.

Next step is to plan implementation of leaflet

**Comment by subhankarb** [comment]
FIXED.
We can check it on https://staging.datapackaged.com/core/ex-geojson

**Comment by rufuspollock** [comment]
FIXED.
